config/testconfig.go

```diff
@@ -39,3 +49,380 @@ type TestConfig struct {
+		}
+		if c.Logging.Grafana == nil {
+			c.Logging.Grafana = &GrafanaConfig{}
+		}
+		bearerToken := grafanaBearerToken.(string)
+		logger.Debug().Msgf("Using E2E_TEST_GRAFANA_BEARER_TOKEN env var to override Logging.Grafana.BearerToken")
+		c.Logging.Grafana.BearerToken = &bearerToken
+	}
+
+	pyroscopeServerUrl, err := readEnvVarValue("E2E_TEST_PYROSCOPE_SERVER_URL", String)
+	if err != nil {
+		return err
+	}
+	if pyroscopeServerUrl != nil && pyroscopeServerUrl.(string) != "" {
+		if c.Pyroscope == nil {
+			c.Pyroscope = &PyroscopeConfig{}
+		}
+		serverUrl := pyroscopeServerUrl.(string)
+		logger.Debug().Msgf("Using E2E_TEST_PYROSCOPE_SERVER_URL env var to override Pyroscope.ServerUrl")
+		c.Pyroscope.ServerUrl = &serverUrl
+	}
+
+	pyroscopeKey, err := readEnvVarValue("E2E_TEST_PYROSCOPE_KEY", String)
+	if err != nil {
+		return err
+	}
+	if pyroscopeKey != nil && pyroscopeKey.(string) != "" {
+		if c.Pyroscope == nil {
+			c.Pyroscope = &PyroscopeConfig{}
+		}
+		key := pyroscopeKey.(string)
+		logger.Debug().Msgf("Using E2E_TEST_PYROSCOPE_KEY env var to override Pyroscope.Key")
+		c.Pyroscope.Key = &key
+	}
+
+	pyroscopeEnvironment, err := readEnvVarValue("E2E_TEST_PYROSCOPE_ENVIRONMENT", String)
+	if err != nil {
+		return err
+	}
+	if pyroscopeEnvironment != nil && pyroscopeEnvironment.(string) != "" {
+		if c.Pyroscope == nil {
+			c.Pyroscope = &PyroscopeConfig{}
+		}
+		environment := pyroscopeEnvironment.(string)
+		logger.Debug().Msgf("Using E2E_TEST_PYROSCOPE_ENVIRONMENT env var to override Pyroscope.Environment")
+		c.Pyroscope.Environment = &environment
+	}
+
+	pyroscopeEnabled, err := readEnvVarValue("E2E_TEST_PYROSCOPE_ENABLED", Boolean)
+	if err != nil {
+		return err
+	}
+	if pyroscopeEnabled != nil {
+		if c.Pyroscope == nil {
+			c.Pyroscope = &PyroscopeConfig{}
+		}
+		enabled := pyroscopeEnabled.(bool)
+		logger.Debug().Msgf("Using E2E_TEST_PYROSCOPE_ENABLED env var to override Pyroscope.Enabled")
+		c.Pyroscope.Enabled = &enabled
+	}
+
+	return nil
+}
+
+// loadEnvVarGroupedMap scans all environment variables, matches them against
+// a specified pattern, and returns a map of grouped values based on the pattern.
+// The grouping is defined by the first capture group of the regex.
+func loadEnvVarGroupedMap(pattern string) map[string][]string {
+	logger := logging.GetTestLogger(nil)
+	re := regexp.MustCompile(pattern)
+	groupedVars := make(map[string][]string)
+	for _, env := range os.Environ() {
+		pair := strings.SplitN(env, "=", 2)
+		if len(pair) != 2 {
+			continue
+		}
+		key, value := pair[0], pair[1]
+		matches := re.FindStringSubmatch(key)
+		if len(matches) > 1 && value != "" {
+			group := matches[1] // Use the first capture group for grouping
+			groupedVars[group] = append(groupedVars[group], value)
+			logger.Debug().Msgf("Will override test config from env var '%s'", key)
+		}
+	}
+	return groupedVars
+}
+
+func loadEnvVarSingleMap(pattern string) map[string]string {
+	logger := logging.GetTestLogger(nil)
```

Comment:
Do you know if we need a test logger here or if we don't? I am not sure it can properly work with `nil`
Also, please check other places.

Reply:
@skudasov yeah, we need it to show logs about overrides. Also, it works fine with testing.T nil